### DISCUSSION TITLE
access to protocols and classes in extension

### DISCUSSION THREAD

**Discussion by Bjorninn1** [airbnb/lottie-ios#1999] [General]
Hello. 
I just wanted to ask if someone could point me in some direction with adding functionalities to the library.

I have tried the following:

```
import Lottie
extension RootAnimationLayer {
    
}
//error: Cannot find type 'RootAnimationLayer' in scope

```

I see how the protocols are not public and therefore not accessible from outside the library. Also there are private variables so I can't access them.

I have managed to add a new function to the library itself. But this requires changing things inside the pod.

Is there maybe a better way of doing this to prevent having to add this code to each new version I may download in the future?

## Which Version of Lottie are you using?

Lottie 4.1.3

**Comment by calda** [comment]
These types are internal implementation details and not something that we would expose publicly. 

What are you trying to change? It would be easiest to just make your changes as a pull request to the main lottie-ios repo, unless the changes are really specific / not generalizable.

**Reply by Bjorninn1** [reply]
Great, thank you very much. It's a method that returns a list of all layers. I have now created a pull request.
